File: fib.go
package main

import "fmt"
import "strconv"
import "os"

func fib(n int) int {
    if (n == 1) {
        return 1
    }
    if (n == 2) {
        return 1
    }

    return fib(n-1) + fib(n-2)
}

func main() {
    getnum := os.Args[1]
    n, err := strconv.Atoi(getnum)
    fmt.Println("res = ", fib(n))

    if (err != nil) {
        fmt.Println(err)
    }
}

File: naive/fib.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdint.h>

int fib(int64_t n);

int main(int argc, char *argv[]) {

    int64_t n;
    n = atoi(argv[1]);

    printf("res = %li\n", fib(n));
    
    return 0;

}

int fib(int64_t n) {
    if (n == 1) {
        return 1;
    }
    if (n == 2) {
        return 1;
    }

    return fib(n-1) + fib(n-2);
}